apis/v1alpha2/shared_types.go

```diff
@@ -244,7 +244,22 @@ type RouteStatus struct {
 	Gateways []RouteGatewayStatus `json:"gateways"`
 }
 
-// Hostname is used to specify a hostname that should be matched.
+// Hostname is the fully qualified domain name of a network host, as defined
+// by RFC 3986. Note the following deviations from the "host" part of the
+// URI as defined in the RFC:
+//
+// 1. IP literals are not allowed.
```

Comment:
Per RFC 6066, the name in a SNI message is not permitted to be a IP address literal.

Reply:
Hostname is used for more than just SNI though. The `Host` header allows IP literal?

Reply:
Yeh, but it would be confusing to have different kinds of hostname with different semantics. IIRC the idea is to have consisent hostname semantics across the API.

Reply:
I don't know.. I see your point but it feels a bit like hiding the complexity of the realities of the internet behind a overly simplified API. A bit like Ingress. Would be interested in others opinions here.